cli/exchange/service.go

```diff
@@ -67,11 +74,44 @@ func ServiceList(credOrg, userPw, service string, namesOnly bool) {
 		if httpCode == 404 && service != "" {
 			cliutils.Fatal(cliutils.NOT_FOUND, msgPrinter.Sprintf("service '%s' not found in org %s", service, svcOrg))
 		}
-		jsonBytes, err := json.MarshalIndent(services.Services, "", cliutils.JSON_INDENT)
+
+		exchServices := services.Services
+		var clusterDeployment string
+		var svcId string
+		for sId, s := range exchServices {
+			// when only one service is specified, save the ClusterDeployment and service id for later use
+			if service != "" {
+				clusterDeployment = s.ClusterDeployment
+				svcId = sId
+			}
+
+			// only display 100 charactors for ClusterDeployment because it is usually very long
+			if s.ClusterDeployment != "" && len(s.ClusterDeployment) > 100 {
+				if service != "" && !namesOnly {
+					// if user specify a service name and -l is specified, then display all of ClusterDeployment
+					// this will give user a way to examine all of it.
+					continue
+				}
+				s_copy := exchange.ServiceDefinition(s)
+				s_copy.ClusterDeployment = s.ClusterDeployment[0:100] + "..."
+				exchServices[sId] = s_copy
+			}
+		}
+		jsonBytes, err := json.MarshalIndent(exchServices, "", cliutils.JSON_INDENT)
 		if err != nil {
 			cliutils.Fatal(cliutils.JSON_PARSING_ERROR, msgPrinter.Sprintf("failed to marshal 'hzn exchange service list' output: %v", err))
 		}
 		fmt.Println(string(jsonBytes))
+
+		// save the kube operator yaml archive to file if filePath is specified and one service is specified
+		if filePath != "" {
+			if clusterDeployment == "" {
+				msgPrinter.Printf("Skip file saving because the clusterDepolyment attribute is empty for this service.")
```

Comment:
change "Skip file saving because" to "Ignoring -f because"

Reply:
done